src/ws/cockpit-desktop.in

```diff
@@ -165,23 +165,32 @@ esac
 
 detect_browser
 
+if test -f /.flatpak-info; then
+    ssh_cmd='flatpak-spawn --host ssh'
+    unshare_shell='/bin/bash -c'
+else
+    ssh_cmd='ssh'
+    unshare_shell='unshare --user --map-root-user --net /bin/bash -c'
+fi
+
 # start the bridge; this needs to run in the normal user session/namespace
-coproc ${2:+ssh "$2"} cockpit-bridge
+coproc ${2:+$ssh_cmd "$2"} cockpit-bridge
 trap "kill $COPROC_PID; wait $COPROC_PID || true" EXIT INT QUIT PIPE
 
 # start ws and browser in a detached network namespace
 SCRIPT='
-set -eu
 # new namespaces have lo down by default
-ip link set lo up >&2
+if ! test -f /.flatpak-info; then
+    ip link set lo up >&2
+fi
 
 # start browser in a temporary home dir, so that it does not interfere with your real one
 export BROWSER_HOME=$(mktemp --directory --tmpdir cockpit.desktop.XXXXXX)
 
 # forward parent stdin and stdout (from bridge) to cockpit-ws
 # it pretty well does not matter which port we use in our own namespace, so use standard http
 # disable /etc/cockpit/
-XDG_CONFIG_DIRS="$BROWSER_HOME" COCKPIT_SUPERUSER="pkexec" '${COCKPIT_WS:-@libexecdir@/cockpit-ws}' -p 80 -a 127.0.0.90 --local-session=- <&0 >&1 &
+XDG_CONFIG_DIRS="$BROWSER_HOME" COCKPIT_SUPERUSER="pkexec" '${COCKPIT_WS:-@libexecdir@/cockpit-ws}' -p 9090 -a 127.0.0.90 --local-session=- <&0 >&1 &
```

Comment:
If flatpaks have their own  network namespace, this ought to work? (or is it because the process does not run as virtual root there?) If this has to be changed, please also adjust the comment above.

Reply:
Again, this is there by accident.  I'll revert it.  cockpit-client (choosing host via UI and having ssh spawned later) and cockpit-desktop (ssh connection already at the start, from commandline arguments) ended up being too structurally different.